feat(icon): new component and example usage in menu

alexciesielski opened the pull request:
### Feature Requests

As described [in this discussion](https://github.com/orgs/primefaces/discussions/1994) I want to use a different set of icons in the `menu` component, without having to override the whole item template.

I just went ahead and implemented the basics without waiting for confirmation, as it was such an easy thing to implement.

If accepted I can go ahead and add a documentation page for the `Icon` component and additionally refactor other components to make use of it, instead of directly using `<i class ....>`.

SoyDiego commented:
> ### Feature Requests
> 
> As described [in this discussion](https://github.com/orgs/primefaces/discussions/1994) I want to use a different set of icons in the `menu` component, without having to override the whole item template.
> 
> I just went ahead and implemented the basics without waiting for confirmation, as it was such an easy thing to implement.
> 
> If accepted I can go ahead and add a documentation page for the `Icon` component and additionally refactor other components to make use of it, instead of directly using `<i class ....>`.

Thanks for your contribution, could you add a stackblitz or screenshots to see your implementation?

Thanks!

alexciesielski commented:
Well, on the surface nothing changes. By default it is supposed to look exactly the same as it does right now. 

It only looks different if you provide your own IconComponent. 

Check out the unit test I wrote to see the usage. 

Also feel free to ask questions if anything is unclear.

mertsincan commented:
Thanks a lot for your contribution! Could you please update your PR and create an issue? Then, we'll review it asap.

Best Regards,

mertsincan commented:
For now, closed! If you have an update from your side, please reopen this PR. Thanks a lot for your contribution!

alexciesielski commented:
Hey, sorry it took so long.

It seems I'm not allowed to reopen this PR. Anyway, I have updated the branch against latest master (v18 it seems) and added an issue here:

https://github.com/primefaces/primeng/issues/16974

imaksp commented on src/app/components/icon/icon.ts:
```diff
@@ -0,0 +1,38 @@
+    selector: 'p-icon',
+    template: `
+        @if(!iconComponentClass) {
+        <i [ngClass]="icon" [ngStyle]="iconStyle"></i>
```
ngClass & ngStyle might be deprecated, [class] & [style] are preferred.

alexciesielski replied:
Do you have a link to support this claim that they are deprecated? The official docs don't say anything about that fact https://angular.dev/api/common/NgClass

imaksp replied:
sure, see this for details:
https://github.com/angular/angular/pull/58860
its also mentioned in upcoming style guide.

alexciesielski replied:
Interesting thanks. I will change it.
